app/lib/bk/compat/parsers/bitrise/workflows.rb

```diff
@@ -7,6 +7,16 @@ class Bitrise
       private
 
       def parse_workflow(wf_name, wf_config)
+        bk_steps = process_workflow_steps(wf_name, wf_config)
+
+        BK::Compat::GroupStep.new(
+          label: wf_name,
+          key: wf_name,
+          steps: [bk_steps]
+        )
+      end
+
+      def process_workflow_steps(wf_name, wf_config)
```

Comment:
this change means most examples end up with groups with single steps, which kind of defeat the purpose of the group. Or is there a reason for this that I'm missing?

Reply:
Good question @toote - I probably should have included this in the PR message - effectively this'll be the precursor to translating both `before_run` and `after_run` [params](https://devcenter.bitrise.io/en/references/workflows-reference.html) that a `workflow` can have defined alongside its `steps` (its about 6 examples down). Effectively, the two options run the `workflow` steps pre/post the `steps` of the actual workflow - thinking more to have these defined in the eventual translation as appending pre/post `CommandStep`s if we encounter either/or.

What do you think of this approach?

Reply:
from what I've read, the `before_run` and `after_run` parameters define steps that need to run in the same execution environment as the rest of the workflow, which in BuildKite it translates to running inside the same step, so adding a GroupStep is not necessary as they would be commands within the same step anyways

Reply:
Thats fair - we'll assess this in due time but for now, I've wrapped everything back to singular `CommandStep`s in the latest commit!